awscrt/http.py

```diff
@@ -144,31 +143,94 @@ def on_incoming_headers_received(headers, response_code, has_body):
 
         return request
 
+
+'''
+TODO: Server Connection class,and configure the server connection.
+'''
+
+
+class HttpServer(object):
+    """
+    Represents an Http server. Everything in this class is non-blocking.
+    """
+    __slots__ = ('_bootstrap', '_tls_connection_options', '_on_incoming_connection',
+                 '_on_destroy_complete', '_native_handle', '_destroy_complete')
+
+    def __init__(self, bootstrap, host_name, port, socket_options, on_incoming_connection, tls_connection_options=None):
+        """
+        Create a new server listener, binding to the host_name and port. 
+        When a new connection is received, the on_incoming_connection cb will be fired, a new ServerConnection obj will be created.
+        The aws_py_http_connection_configure_server need to be called from the callback to configure the ServerConnection 
+        @param socket_options: awscrt.io.SocketOptions for the server's listening socket. Required
+        @param on_incoming_connection: Callback with signature (connection: HttpConnection.native_handle, error_code: int) Required
+        @param bootstrap: awscrt.io.ServerBootstrap. Required
+        @param tls_connection_options: awscrt.io.TlsConnectionOptions, for TLS connection
+        """
+        assert isinstance(bootstrap, ServerBootstrap)
+        assert tls_connection_options is None or isinstance(tls_connection_options, TlsConnectionOptions)
+        assert host_name is not None
+        assert port is not None
+        assert isinstance(socket_options, SocketOptions)
+        assert on_incoming_connection is not None
+        for slot in self.__slots__:
+            setattr(self, slot, None)
+
+        def on_destroy_complete(server_native_handle):
+            self._destroy_complete.set_result(True)
+
+        self._bootstrap = bootstrap
+        self._tls_connection_options = tls_connection_options
+        self._on_incoming_connection = on_incoming_connection
+        self._on_destroy_complete = on_destroy_complete
+        self._native_handle = None
+        self._destroy_complete = Future()
+
+        if tls_connection_options is not None:
+            internal_conn_options_handle = tls_connection_options._internal_tls_conn_options
+        else:
+            internal_conn_options_handle = None
+
+        self._native_handle = _aws_crt_python.aws_py_http_server_create(
+            bootstrap._internal_bootstrap, on_incoming_connection, on_destroy_complete, host_name, port, socket_options,
+            internal_conn_options_handle)
+
+    def close(self):
+        """
+        close the server, no more connections will be accepted, a future object will be returned, and when the close process finishes
+        the future result or exception will be set.
+        """
+        try:
+            _aws_crt_python.aws_py_http_server_release(self._native_handle)
+        except Exception as e:
+            self._destroy_complete.set_exception(e)
```

Comment:
I don't think we should use try/except here. We should probably just let the exception go uncaught and let the user handle it. The only cause for exception is this function somehow running on an object that isn't actually a Server capsule, right?